From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Manish Bhardwaj <mbhardwaj@nvidia.com>
Date: Wed, 27 Aug 2025 03:12:43 +0000
Subject: [PATCH 102/243] NVIDIA: SAUCE: tegra_hv: remove
 tegra_hv_ivc_convert_cookie API

tegra_hv_ivc_convert_cookie API is not being used
and is there lying as dead code.

Bug 5433066

Change-Id: Ib6b7a151f6da2eb7c4fbab1f632c87eb16397d69
Signed-off-by: Manish Bhardwaj <mbhardwaj@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3449492
Reviewed-by: svcacv <svcacv@nvidia.com>
Reviewed-by: Sandeep Trasi <strasi@nvidia.com>
Reviewed-by: Suresh Venkatachalam <skathirampat@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
(cherry picked from commit bb8edf1919e49386ff52e5dd3849a24e2f0d88f4)

diff --git a/drivers/virt/tegra/tegra_hv.c b/drivers/virt/tegra/tegra_hv.c
index 869380d307532..9b1be4d1db2b5 100644
--- a/drivers/virt/tegra/tegra_hv.c
+++ b/drivers/virt/tegra/tegra_hv.c
@@ -1033,12 +1033,6 @@ int tegra_hv_ivc_read_advance(struct tegra_hv_ivc_cookie *ivck)
 }
 EXPORT_SYMBOL(tegra_hv_ivc_read_advance);
 
-struct tegra_ivc *tegra_hv_ivc_convert_cookie(struct tegra_hv_ivc_cookie *ivck)
-{
-	return &cookie_to_ivc_dev(ivck)->ivc;
-}
-EXPORT_SYMBOL(tegra_hv_ivc_convert_cookie);
-
 struct tegra_hv_ivm_cookie *tegra_hv_mempool_reserve(unsigned int id)
 {
 	uint32_t i;
diff --git a/include/soc/tegra/virt/hv-ivc.h b/include/soc/tegra/virt/hv-ivc.h
index dab9b26bc39a7..35fdb9876e1fd 100644
--- a/include/soc/tegra/virt/hv-ivc.h
+++ b/include/soc/tegra/virt/hv-ivc.h
@@ -306,6 +306,4 @@ int tegra_hv_ivc_get_info(struct tegra_hv_ivc_cookie *ivck, uint64_t *pa,
  */
 void tegra_hv_ivc_notify(struct tegra_hv_ivc_cookie *ivck);
 
-struct tegra_ivc *tegra_hv_ivc_convert_cookie(struct tegra_hv_ivc_cookie *ivck);
-
 #endif /* __TEGRA_HV_IVC_H */
-- 
2.43.0

